compiler/optimizer/LocalOpts.cpp

```diff
@@ -3548,6 +3548,27 @@ int32_t TR_EliminateRedundantGotos::process(TR::TreeTop *startTree, TR::TreeTop
             !performTransformation(comp(), "%sEliminating goto at the end of block_%d with BBStart %p\n", optDetailString(), block->getNumber(), block->getEntry()->getNode()))
         continue;
 
+      // Blocks could be removed when the exception edges are removed from the block
+      // The IN edges saved in fixablePreds need to be updated
+      for (auto previousEdge = fixablePreds.before_begin(), currentEdge = fixablePreds.begin();
+           currentEdge != fixablePreds.end();
+           ++previousEdge, ++currentEdge)
+         {
+         TR::CFGEdge* current = *currentEdge;
+         TR::Block *from = toBlock(current->getFrom());
+         TR::Block *to = toBlock(current->getTo());
+
+         if (!from->isValid() || !to->isValid())
+            {
+            if (trace())
+               traceMsg(comp(), "%s: remove edge block_%d (isValid %d) -> block_%d (isValid %d) from fixablePreds\n", __FUNCTION__,
+                  from->getNumber(), from->isValid(), to->getNumber(), to->isValid());
+
+            fixablePreds.erase_after(previousEdge);
+            currentEdge = previousEdge;
```

Comment:
Once an edge is removed, we'll have `currentEdge == previousEdge` for the remainder of the loop, so any later attempt to remove another edge won't work properly. I think the iterator updates should come out of the `for` step so that we can avoid updating `previousEdge` after running this case, e.g.
```
auto previousEdge = fixablePreds.before_begin();
auto currentEdge = fixablePreds.begin();
while (currentEdge != fixablePreds.end())
   {
   if (...bad...)
      {
      currentEdge = fixablePreds.erase_after(previousEdge);
      }
   else
      {
      ++previousEdge;
      ++currentEdge;
      }
   }
```